File: app/src/main/java/eu/sesma/dagger/ui/main/MainActivity.java
package eu.sesma.dagger.ui.main;

import eu.sesma.dagger.R;
import eu.sesma.dagger.ui.BaseActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import javax.inject.Inject;

import dagger.android.AndroidInjection;

public class MainActivity extends BaseActivity {

    @Inject
    MainPresenter presenter;

    View.OnClickListener listener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            presenter.onClick();
        }
    };

    @Override
    public void onCreate(Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        presenter.initialize();
        setTitle(presenter.getCollaboratorVersion());
        ((Button) findViewById(R.id.button)).setOnClickListener(listener);
    }
}

File: app/src/main/java/eu/sesma/dagger/ui/main/MainActivity.kt
package eu.sesma.dagger.ui.main

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.TextView
import eu.sesma.dagger.R
import eu.sesma.dagger.di.DaggerActivityComponent
import eu.sesma.dagger.platform.ActivityModule
import eu.sesma.dagger.platform.AndroidApplication
import eu.sesma.main.LibraryMain
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var presenter: MainPresenter

    private val listener: View.OnClickListener = View.OnClickListener { presenter.onClick() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        DaggerActivityComponent.builder()
                .applicationComponent((application as AndroidApplication).applicationComponent)
                .activityModule(ActivityModule(this))
                .build().inject(this)


        presenter.initialize()
        title = presenter.appCollaboratorVersion
        (findViewById<View>(R.id.button) as Button).setOnClickListener(listener)

        val libraryMain = LibraryMain()
        libraryMain.initialize()

        findViewById<TextView>(R.id.textView2).text = presenter.libraryCollaboratorGreet

        findViewById<TextView>(R.id.textView3).text = presenter.coreCollaboratorGreet
    }
}
